usecase/parser/v0_43_0/ibcmsg/ibcmsg.go

```diff
@@ -0,0 +1,150 @@
+			parserParams.MsgCommonParams,
+
+			msgRecvPacketParams,
+		)}
+	}
+
+	log := utils.NewParsedTxsResultLog(&parserParams.TxsResult.Log[parserParams.MsgIndex])
+
+	recvPacketEvent := log.GetEventByType("recv_packet")
+	if recvPacketEvent == nil {
+		panic("missing `recv_packet` event in TxsResult log")
+	}
+
+	fungibleTokenPacketEvent := log.GetEventByType("fungible_token_packet")
+	if fungibleTokenPacketEvent == nil {
+		// Note: this could happen when the packet is already relayed.
+		// https://github.com/cosmos/ibc-go/blob/760d15a3a55397678abe311b7f65203b2e8437d6/modules/core/04-channel/keeper/packet.go#L239
+		// https://github.com/cosmos/ibc-go/blob/760d15a3a55397678abe311b7f65203b2e8437d6/modules/core/keeper/msg_server.go#L508
+
+		packetAck := ibc_model.MsgRecvPacketPacketAck{
+			MaybeError: primptr.String("missing `fungible_token_packet` event in TxsResult log, this could happen when the packet is already relayed"),
+		}
+
+		msgRecvPacketParams := ibc_model.MsgRecvPacketParams{
+			RawMsgRecvPacket: rawMsg,
+
+			Application: "transfer",
+			MessageType: "MsgTransfer",
+			MaybeFungibleTokenPacketData: &ibc_model.MsgRecvPacketFungibleTokenPacketData{
+				FungibleTokenPacketData: rawFungibleTokenPacketData,
+				Success:                 false,
+			},
+
+			PacketSequence: typeconv.MustAtou64(recvPacketEvent.MustGetAttributeByKey("packet_sequence")),
+			// TODO: Remove this when IBCChannel projection no longer relies on packetAck.MaybeError to check if MsgRecvPacket is success or not
+			PacketAck: packetAck,
+		}
+
+		return []command.Command{command_usecase.NewCreateMsgIBCRecvPacket(
+			parserParams.MsgCommonParams,
+
+			msgRecvPacketParams,
+		)}
+	}
+
+	var maybeDenominationTrace *ibc_model.MsgRecvPacketFungibleTokenDenominationTrace
+	denominationTraceEvent := log.GetEventByType("denomination_trace")
+	if denominationTraceEvent != nil {
+		maybeDenominationTrace = &ibc_model.MsgRecvPacketFungibleTokenDenominationTrace{
+			Hash:  denominationTraceEvent.MustGetAttributeByKey("trace_hash"),
+			Denom: denominationTraceEvent.MustGetAttributeByKey("denom"),
+		}
+	}
+
+	writeAckEvent := log.GetEventByType("write_acknowledgement")
+	if writeAckEvent == nil {
+		panic("missing `write_acknowledgement` event in TxsResult log")
+	}
+	var packetAck ibc_model.MsgRecvPacketPacketAck
+	json.MustUnmarshalFromString(writeAckEvent.MustGetAttributeByKey("packet_ack"), &packetAck)
+
+	msgRecvPacketParams := ibc_model.MsgRecvPacketParams{
+		RawMsgRecvPacket: rawMsg,
+
+		Application: "transfer",
+		MessageType: "MsgTransfer",
+		MaybeFungibleTokenPacketData: &ibc_model.MsgRecvPacketFungibleTokenPacketData{
+			FungibleTokenPacketData: rawFungibleTokenPacketData,
+			// success value inverted bug: https://github.com/cosmos/cosmos-sdk/pull/9640
+			// TODO: after fixing this, please also update `projection/ibc_channel/ibc_channel.go -> HandleEvents()`
+			//			 when event type is `MsgIBCRecvPacket`.
```

Comment:
Should we resolve these TODO as we have in-inverted the `success` value now?

Reply:
sure

Reply:
yes